portal/client-services/org.wso2.is.portal.user.client.api/src/main/java/org/wso2/is/portal/user/client/api/IdentityStoreClientService.java

```diff
@@ -141,4 +143,11 @@ UUFUser addUser(Map<String, String> userClaims, Map<String, String> credentials,
      */
     List<UUFUser> listUsers(String claimUri, String claimValue, int offset, int length,
                             String domainName) throws UserPortalUIException;
+
+    List<UserListBean> getFilteredList(int offset, int length, String claimURI,
```

Comment:
@omindu add java doc comments & method name reflect the idea of getting already filtered list. method name should be something like listUsersByFilter